imf/distributions.py

```diff
@@ -156,6 +172,22 @@ def rvs(self, N):
                 ret.append(self.pows[ii].rvs(Ns[ii]))
         return np.concatenate(ret)
 
+    def ppf(self, x0):
+        x = np.asarray(x0)
+        x1 =np.atleast_1d(x)
+        assert(x1.min()>=0)
+        assert(x1.max()<=1)
+        edges = np.r_[[0],np.cumsum(self.weights)]
+        # edges of powerlaw in CDF scale from 0 to 1
+        pos = np.digitize(x1, edges) # bin positions, 1 is the leftmost 
+        left = edges[pos-1]
+        w = self.weights[pos-1]
+        x2 = np.clip((x1-left)/w,0,1) # mapping to 0,1 on the segment
+        ret =np.zeros_like(x1)
+        for ii in range(x.size):
+            ret[ii] = self.pows[pos[ii]-1].ppf(x2[ii])
```

Comment:
this might be more efficiently implemented as a list comprehension

Reply:
I agree that the line is not the most efficient, but I don't think there is a way around that -- multiple points may correspond to different segments. Regarding the list comprehension, to my knowledge that'll be just stylistic difference. I'm not expecting any performance difference.
I can imagine a faster version (if input is long) is to group by segment. But I thought I first implement a trivial version and if it is ever a problem on can optimize it later.

Reply:
the list comprehension is a little more than a stylistic difference, but it's unclear whether there would be a performance benefit in this case, so I agree, leave it as is.